src/modules/io.c

```diff
@@ -308,3 +308,25 @@ internal void
 FILESYSTEM_getBasePath(WrenVM* vm) {
   wrenSetSlotString(vm, 0, BASEPATH_get());
 }
+
+internal void
+FILESYSTEM_createDirectory(WrenVM *vm) {
+  const char* path = wrenGetSlotString(vm, 1);
+  const char* fullPath;
+  mode_t mode = 0777;
+  if (path[0] != '/') {
+    char* base = BASEPATH_get();
+    fullPath = malloc(strlen(base)+strlen(path)+1);
+    strcpy((char*)fullPath, base);
+    strcat((char*)fullPath, path);
+  } else {
+    fullPath = path;
+  }
```

Comment:
I realise that this is my own fault, but it would be good to pull the path-resolving logic out into its own method, so that we aren't copy-pasting it between functions. I'd suggest putting this method in `src/io.c`, as:

```c
char* resolvePath(char* partialPath, bool* shouldFree) {
    const char* fullPath;
    mode_t mode = 0777;
    if (partialPath[0] != '/') {
      char* base = BASEPATH_get();
      fullPath = malloc(strlen(base)+strlen(partialPath)+1);
      strcpy((char*)fullPath, base);
      strcat((char*)fullPath, partialPath);
      if (shouldFree != NULL) {
        *shouldFree = true;
      }
    } else {
      fullPath = partialPath;
      if (shouldFree != NULL) {
        *shouldFree = false;
      }
    }

    return fullPath;
}
```

Then we can call it as 
```c
bool shouldFree;
char* fullPath = resolvePath(path, &shouldFree);
// Use the full path for something here.
// For example:
// int result = mkdirp(fullPath, mode);
if (shouldFree) {
  free((void*)fullPath);
}
```